include/boost/json/impl/pointer.ipp

```diff
@@ -0,0 +1,329 @@
+            if ( result <= max_size - d )
+            {
+                result += d;
+                continue;
+            }
+        }
+
+        ec = error::token_overflow;
+        return {};
+    }
+    return result;
+}
+
+pointer_token
+get_token(
+    string_view::const_iterator b,
+    string_view::const_iterator e,
+    error_code& ec) noexcept
+{
+    auto const start = b;
+    for (; b < e; ++b)
+    {
+        char const c = *b;
+        if ( '/' == c )
+        {
+            break;
+        }
+
+        if ( '~' == c )
+        {
+            if ( ++b == e )
+            {
+                ec = error::invalid_escape;
+                break;
+            }
+
+            switch (*b)
+            {
+            case '0': // fall through
+            case '1':
+                // valid escape sequence
+                continue;
+            default:
+                ec = error::invalid_escape;
+                break;
+            }
+            break;
+        }
+    }
+
+    return pointer_token(start, b);
+}
+
+value*
+if_contains_token(object const& obj, pointer_token token)
+{
+    if(obj.empty())
+    {
+        return nullptr;
+    }
+
+    auto const it = detail::find_in_object(obj, token).first;
+    if(!it)
+    {
+        return nullptr;
+    }
+
+    return &it->value();
+}
+
+template <class Value>
+Value*
+find_pointer(Value& root, pointer ptr, error_code& ec) noexcept
+{
+    ec.clear();
+    string_view const ptr_str = ptr.string();
+    Value* result = &root;
+    auto cur = ptr_str.begin();
+    while ( ptr_str.end() != cur )
+    {
+        if ( *cur++ != '/' )
+        {
+            ec = error::missing_slash;
+            return nullptr;
+        }
+
+        if ( auto object = result->if_object() )
+        {
+            auto const token = get_token(cur, ptr_str.end(), ec);
+            if ( ec )
+            {
+                return nullptr;
+            }
+
+            result = detail::if_contains_token(*object, token);
+            cur = token.end().base();
+        }
+        else if ( auto array = result->if_array() )
+        {
+            std::size_t index = detail::parse_number_token(cur, ptr_str.end(),
+                    ec);
+            if ( ec )
+            {
+                return nullptr;
+            }
+
+            result = array->if_contains(index);
+        }
+        else
+        {
+            ec = error::value_is_scalar;
+            return nullptr;
+        }
+
+        if ( !result )
+        {
+            ec = error::not_found;
+            return nullptr;
+        }
+    }
+
+    BOOST_ASSERT(result);
+    return result;
+}
+
+template <class Value>
+Value&
+at_pointer(Value& root, pointer ptr)
+{
+    error_code ec;
+    auto const found = find_pointer(root, ptr, ec);
+    if ( !found )
+    {
+        detail::throw_system_error(ec, BOOST_JSON_SOURCE_POS);
+    }
+    return *found;
+}
+
+} // namespace detail
+
+value const&
+value::at(pointer ptr) const
```

Comment:
You put member function definitions for `value` in pointer.ipp?

Reply:
I thought it would be reasonable to put all code related to Pointer into a separate file. But if you think it belongs in `value.ipp`, then I'll move it.